src/xdotoolify.js

```diff
@@ -326,6 +326,45 @@ _Xdotoolify.prototype.autoType = function(selector, text, relpos, timeout) {
   }
   return this;
 };
+
+_Xdotoolify.prototype.sleepUntil = function(predicate, ms, immediate) {
+  this.operations.push({
+    type: 'sleepUntil',
+    predicate: predicate || function() {return new Promise(function(resolve) {
+      resolve(true);
+    })},
+    ms: ms || 1000,
+    immediate: typeof immediate === 'undefined' ? true : immediate
+  });
+
+  return this;
+}
+
+const _sleepUntil = async function(predicate, ms, immediate) {
+  return new Promise(async function(resolve) {
```

Comment:
i don't think you need to wrap this in a `Promise(`

Reply:
Fixed